cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -235,6 +235,9 @@ const pauseFileName = "PAUSE"
 func (s *Server) Run(queue *Queue) {
 	removeIncompleteShards(s.IndexDir)
 
+	// Protect the index directory from concurrent access of builder and cleanup.
+	muIndexDir := sync.Mutex{}
```

Comment:
can be cherry-picked into own commit for master. haoppy for you to do that and push directly to minimize this PR.

Reply:
done